sympy/printing/printer.py

```diff
@@ -338,15 +341,81 @@ def emptyPrinter(self, expr):
         return str(expr)
 
     def _as_ordered_terms(self, expr, order=None):
-        """A compatibility function for ordering terms in Add. """
+        """A compatibility function for ordering terms in Add.
+
+        When ``order`` is neither 'old' nor 'none', this makes printing
+        decisions based on the assumption `is_real` and evaluation for
+        numeric factors in terms of an expr if `is_real` is not None.
+
+        If this is not desired then the code has to be replaced with
+        something that gives mathematically expected ordering for terms.
+        """
         order = order or self.order
 
         if order == 'old':
             return sorted(Add.make_args(expr), key=cmp_to_key(Basic._compare_pretty))
         elif order == 'none':
+            if expr.is_Add:
+                from sympy.core.numbers import Number, NumberSymbol
+                from sympy.core.mul import Mul
+                # Spot the special case of Add(Number, Mul(Number, expr)) with the
+                # first number positive and the second number negative
+                key = lambda x:not isinstance(x, (Number, NumberSymbol))
+                add_args = sorted(Add.make_args(expr), key=key)
+                if (len(add_args) == 2
+                    and isinstance(add_args[0], (Number, NumberSymbol))
+                    and isinstance(add_args[1], Mul)):
+                    mul_args = sorted(Mul.make_args(add_args[1]), key=key)
+                    if (len(mul_args) == 2
+                            and isinstance(mul_args[0], Number)
+                            and add_args[0].is_positive
+                            and mul_args[0].is_negative):
+                        return add_args
```

Comment:
```suggestion
```

Reply:
@asmeurer , I got that section in the wrong block. But now I am seeing if we can just use the non-evaluative core ordering.